File: src/UE4_Packages/FPerPlatformInt.java
/**
 * 
 */
package UE4_Packages;

/**
 * @author FunGames
 *
 */
public class FPerPlatformInt {
	private boolean cooked;
	private long value;

	public FPerPlatformInt(FArchive Ar) throws ReadException {
		cooked = Ar.readBooleanFromUInt8();
		value = Ar.readUInt32();
	}
	public boolean getCooked() {
		return cooked;
	}
	public long getValue() {
		return value;
	}
}

File: src/main/kotlin/me/fungames/jfortniteparse/ue4/assets/objects/FPerPlatformInt.kt
package me.fungames.jfortniteparse.ue4.assets.objects

import me.fungames.jfortniteparse.ue4.UClass
import me.fungames.jfortniteparse.ue4.reader.FArchive
import me.fungames.jfortniteparse.ue4.writer.FArchiveWriter

@ExperimentalUnsignedTypes
class FPerPlatformInt : UClass {
    var cooked : Boolean
    var value: UInt

    constructor(Ar: FArchive) {
        super.init(Ar)
        cooked = Ar.readFlag()
        value = Ar.readUInt32()
        super.complete(Ar)
    }

    fun serialize(Ar: FArchiveWriter) {
        super.initWrite(Ar)
        Ar.writeFlag(cooked)
        Ar.writeUInt32(value)
        super.completeWrite(Ar)
    }

    constructor(cooked : Boolean, value : UInt) {
        this.cooked = cooked
        this.value = value
    }
}